src/test/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactoryTest.java

```diff
@@ -39,49 +40,42 @@ class IotDataPlaneClientFactoryTest {
     private DeviceConfiguration mockDeviceConfiguration;
     @Mock
     private Context mockContext;
-    @Captor
-    private ArgumentCaptor<ChildChanged> ccCaptor;
 
     private IotDataPlaneClientFactory clientFactory;
 
     @BeforeEach
     void setup() {
         Topic dataEndpointTopic = Topic.of(mockContext, DEVICE_PARAM_IOT_DATA_ENDPOINT, "xxxxxx-ats.iot.us-west-2.amazonaws.com");
-        when(mockDeviceConfiguration.getIotDataEndpoint()).thenReturn(dataEndpointTopic);
+        lenient().when(mockDeviceConfiguration.getIotDataEndpoint()).thenReturn(dataEndpointTopic);
         Topic regionTopic = Topic.of(mockContext, DEVICE_PARAM_AWS_REGION, "us-west-2");
-        when(mockDeviceConfiguration.getAWSRegion()).thenReturn(regionTopic);
-
-        doNothing().when(mockDeviceConfiguration).onAnyChange(ccCaptor.capture());
+        lenient().when(mockDeviceConfiguration.getAWSRegion()).thenReturn(regionTopic);
     }
 
     @Test
-    void GIVEN_bad_config_WHEN_configure_client_THEN_does_not_reconfigure_iot_data_client() {
+    void GIVEN_device_configuration_WHEN_crypto_service_available_THEN_configure_iot_data_client() throws TLSAuthException, IoTDataPlaneClientCreationException {
         clientFactory = new IotDataPlaneClientFactory(mockDeviceConfiguration);
         assertThat(clientFactory.getIotDataPlaneClient(), is(notNullValue()));
-
+        lenient().when(mockDeviceConfiguration.getDeviceIdentityKeyManagers()).thenReturn(new KeyManager[0]);
```

Comment:
nit: is this called or not? Shouldn't need to use `lenient` here

Reply:
This should be updated. And integ tests for these changes will be added in the next PR (unit test is in this PR).